File: movewindeps.py
#!/usr/bin/env python3

from subprocess import Popen, PIPE
from os import system, path
import shlex

OUT = "winbuild/src/main"

def find(file, where):
	command = f"find {where} -name {file}"
	command = shlex.split(command)
	
	c = Popen(command, stdout=PIPE)
	stdout, stderr = [(x.decode() if x is not None else None) for x in c.communicate()]
	if stdout == "":
		#print(f"Could Not Find: '{file}'")
		return None
	else:
		src = stdout.strip()
		#print(f"Found '{file}' at '{src}'")
		return src

for d in open("deps.txt", "r").readlines():
	file, dst = d.strip().split(" ", 1)
	
	if "/usr/" in file:
		src = file
	else:
		src = find(file, "'/usr/x86_64-w64-mingw32/sys-root/mingw/lib' '/usr/x86_64-w64-mingw32/sys-root/mingw/bin'")
	if src != None:
		if not path.exists(f"{OUT}/{dst}"):
			system(f"mkdir -p {OUT}/{dst}")
		system(f"cp {src} {OUT}/{dst}")

system(f"chmod -R 777 {OUT}")

File: movewindeps.pl
#!/usr/bin/perl

my $IN = @ARGV[0] || "deps.win";
my $OUT = @ARGV[1] ||"winbuild5/src/main";

sub find{
	my ($file, $where) = @_;

	my $command = "find $where -name $file";

	my $out = `$command`;

	if(!($out eq "")){
		return $out;
	}
}

open(FH, "<", $IN);

while(<FH>){
	my $data = $_;
	if($data =~ /(.*) (.*)/g){
		my ($file, $dst) = ($1, $2);

		my $src;

		if(substr($file, 0, 1) eq "/"){
			$src = $file;
		}else{
			$src = find $file, "'/usr/x86_64-w64-mingw32/sys-root/mingw/lib' '/usr/x86_64-w64-mingw32/sys-root/mingw/bin'";
		}

		if(! $src eq ""){
			if(! -d "$OUT/$dst"){
				`mkdir -p $OUT/$dst`;
			}

			`cp $src $OUT/$dst`;
		}
	}
}

close(FH);

`chmod -R 777 $OUT`;
